packages/perseus/src/components/sortable.tsx

```diff
@@ -584,15 +576,34 @@ class Sortable extends React.Component<SortableProps, SortableState> {
     }, 20);
 
     render(): React.ReactNode {
-        // We don't render the sortable until KaTeX has fully loaded, in case
-        // the draggables are rendering KaTeX content. This is un-optimal as
-        // we end up loading KaTeX even when we may not need it, however it
-        // helps to ensure that the dimensions of the draggables (and thus the
-        // sortable) will be correct when they render, if their contents are
-        // KaTeX-derived.
-        if (this.props.waitForKatexLoad && !this.state.katex) {
-            return <CircularSpinner />;
+        // The math renderer (KaTeX or MathJax) may be loaded asynchronously
+        // the first time the TeX component is used.
+        //
+        // To minimize layout shift, we display a spinner until our TeX renderer
+        // has successfully rendered a test element. If we didn't do this, the user
+        // might see a sortable with empty cells on first render, and then the math
+        // would pop in a few moments later once the rendering library loaded.
+        if (
+            this.props.waitForTexRendererToLoad &&
+            !this.state.texRendererLoaded
+        ) {
+            const {TeX} = getDependencies();
+            return (
+                <>
+                    <CircularSpinner />
+                    <div style={{display: "none"}}>
+                        <TeX
+                            onRender={() =>
+                                this.setState({texRendererLoaded: true})
+                            }
+                        >
+                            1
+                        </TeX>
+                    </div>
+                </>
+            );
```

Comment:
I'm confused by this.  We render the spinner _and_ the `<TeX>`.  Shouldn't we be switching on the `waitForTexRendererToLoad` and `texRendererLoaded` flags and rendering a spinner else the `TeX`?

Reply:
We're using this TeX component to tell the TeX renderer to load and detect when it has loaded. Once it renders, we set `texRendererLoaded` to `true`.